### PR TITLE
Added Taylor series Using Horner Rule in java,c++,README.md files

### LINKED ISSUE #1129: Taylor series in c++, java and README.md

### PR DESCRIPTION
**Pull request by Anshuvats**
Fixes #1129

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Implementation of Taylor series using Horner Rule in Java, C++.

#### Languages Used:
- C++
- Java

#### Files Added:
- Taylor_series_Repo/README.md
- Taylor_series_Repo/Taylor_Series_Horner_Rule.cpp 
-  Taylor_series_Repo/Taylor_Series_Horner_Rule.java

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Anshuvats** [comment]
Hey, @jainaman224  please review this.

File: Taylor_series_Repo/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+# Taylor Series Using Horner's Rule
+  
+
```
Comment (SakshiShreya) [inline Taylor_series_Repo/README.md]: Remove extra space. Only one new line is needed.

File: Taylor_series_Repo/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+f(x0) = a0 + a1x0 + a2x02 + ... + anx0n
+This can, also, be written as:
+
+f(x0) = a0 + x0(a1 + x0(a2 + x0(a3 + ... + (an-1 + anx0)....)
+
+The algorithm proposed by this rule is based on evaluating the monomials formed above starting from the one in the inner-most parenthesis and move out to evaluate the monomials in the outer parenthesis.
+
+## Algorithm
+
+
```
Comment (SakshiShreya) [inline Taylor_series_Repo/README.md]: Remove extra line

File: Taylor_series_Repo/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+3.Take base case:-if(num==0)then return s
+4.Else creating a formula of expansion using Horner's method like s=1+x*(s/num)
+5.return recursive case
+
+## Time Complexity
+    
+- O(n)
+
+
+
```
Comment (SakshiShreya) [inline Taylor_series_Repo/README.md]: Remove extra lines

File: Taylor_series_Repo/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+## Illustration for above algorithim
+
+
```
Comment (SakshiShreya) [inline Taylor_series_Repo/README.md]: Remove extra line

File: Taylor_series_Repo/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+![GIF](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQ75s7IAgoTnSpbsACLb-dolOUtJZ7MaykTtIsYuNIZgir6joYZ)
+
+
+
+
+
+
```
Comment (SakshiShreya) [inline Taylor_series_Repo/README.md]: Remove extra lines

File: Taylor_series_Repo/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+## Implementation
+
+ 
+
```
Comment (SakshiShreya) [inline Taylor_series_Repo/README.md]: Remove extra lines

File: Taylor_series_Repo/Taylor_Series_Horner_Rule.java [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+import java.util.Scanner;
+public class Test {
+static double s; // Declare static variable
+public static void main (String[] args) {	
+	Scanner sc = new Scanner (System.in);
+	int x = sc.nextInt();
+	int num = sc.nextInt();
+	double result = exponential (x, num);
+	System.out.println (+ result);
```
Comment (SakshiShreya) [inline Taylor_series_Repo/Taylor_Series_Horner_Rule.java]: Replace tabs with 4 spaces in lines 7-11, 16-20.

File: Taylor_series_Repo/Taylor_Series_Horner_Rule.java [hunk 8]
```diff
@@ -0,0 +1,26 @@
+// Taylor series using Horner Rule in Java
+
+import java.util.Scanner;
+public class Test {
+static double s; // Declare static variable
```
Comment (SakshiShreya) [inline Taylor_series_Repo/Taylor_Series_Horner_Rule.java]: Increase one indentation from line 5 to 21.

File: Taylor_series_Repo/Taylor_Series_Horner_Rule.java [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+		return s;
+	}	
+	s = 1 + x * (s / num); // formula of Horner's Rule
+	return exponential (x, num - 1); // recursive case
+}
+}
+
+/* Input:- 2 10
+   Output:- 7.388712522045855
+*/
```
Comment (SakshiShreya) [inline Taylor_series_Repo/Taylor_Series_Horner_Rule.java]: Add a new line at the end of file

**Comment by Anshuvats** [comment]
Hey @SakshiShreya, I have updated files. kindly look into this. If any mistake from me, then I will change this.

File: Taylor_series_Repo/Taylor_Series_Horner_Rule.java [hunk 10]
```diff
@@ -4,23 +4,23 @@
...
 
 //Declare the function for finding exponential
 public static double exponential (int x, int num) {
-	if(num == 0) {  // Base case
-		return s;
-	}	
-	s = 1 + x * (s / num); // formula of Horner's Rule
-	return exponential (x, num - 1); // recursive case
+    if(num == 0) {  // Base case
+      return s;
```
Comment (SakshiShreya) [inline Taylor_series_Repo/Taylor_Series_Horner_Rule.java]: Use 8 spaces instead of 6.

**Comment by Anshuvats** [comment]
@SakshiShreya @jainaman224, please check this. I have updated this.

File: Taylor_series_Repo/Taylor_Series_Horner_Rule.cpp [hunk 11]
```diff
@@ -0,0 +1,29 @@
+// Taylor series using Horner Rule in c++
+
+#include<bits/stdc++.h>
+using namespace std;
+
+// creating exponential function
+double exponential (int x, int num)
+{
+    static double s; // declare static variable
+    if(num == 0)   // Base case
```
Comment (marwahmanbir) [inline Taylor_series_Repo/Taylor_Series_Horner_Rule.cpp]: Add a space after ``if`` and ``for`` throughout the code.

File: Taylor_series_Repo/Taylor_Series_Horner_Rule.cpp [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+        return s;
+    }
+    s = 1 + x * (s / num); // formula of horner's rule
+    return exponential (x, num - 1); // recursive case
+}
+// main method
+int main ()
+{
+    int x, num;
+    cin >> x >> num;
```
Comment (marwahmanbir) [inline Taylor_series_Repo/Taylor_Series_Horner_Rule.cpp]: Space around ``>>`` & ``<<`` is not necessary in ``cin`` & ``cout`` statements.

File: Taylor_series_Repo/Taylor_Series_Horner_Rule.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+    static double s; // Declare static variable
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+        int x = sc.nextInt();
+        int num = sc.nextInt();
+        double result = exponential(x, num);
+        System.out.println(+result);
+    }
+
+    //Declare the function for finding exponential
```
Comment (marwahmanbir) [inline Taylor_series_Repo/Taylor_Series_Horner_Rule.java]: Always add a space after **//** in comments. Rest of the file looks good. :sparkles:

File: Taylor_series_Repo/Taylor_Series_Horner_Rule.cpp (hunk 12 above)
Comment (SakshiShreya) [reply Taylor_series_Repo/Taylor_Series_Horner_Rule.cpp]: `>>` and `<<` are operators. There should be space around them.

Comment (Anshuvats) [reply Taylor_series_Repo/Taylor_Series_Horner_Rule.cpp]: @marwahmanbir, @SakshiShreya, What should I do?

**Pull request closed without merging**